### DISCUSSION TITLE
[Question] Compression for local file transfers

### DISCUSSION THREAD

**Discussion by jb2170** [RsyncProject/rsync#362] [Q&A] [answered]
Hi, sorry I couldn't find this in the documentation so I'll ask:

Does the `-z` (compression) flag have any effect for local file transfers, that is say does `rsync -az foo/ bar/` use compression?

I find myself sometimes doubting whether I should use `-vzaH` or `-vaH` as my muscle-memory go-to default flags, whether it's a local-to-local or local-to-remote setup. It seems a bit superfluous if rsync does use compression for local-to-local file transfers right? It'd be nice to know that it doesn't and I can use `-z` freely without thinking about possible wasted time / energy compressing and decompressing unnecessarily.

Note: I'm not talking about say `rsync -az foo/ localhost:~/bar/`, for which I would guess compression to be used as per any rsync-over-ssh usage that uses `-z`, agnostic of whether the host and target machine are the same.

**Comment by WayneD** [comment, marked as answer]
You're probably wasting cpu on a fast socket or pipe. While it compresses the pipe data on a local transfer, it's not needed.  You can see its effect by requesting `--stats`.

**Reply by jb2170** [reply]
What's the point of compression for a local file transfer? Is there something I'm not thinking about? Why is rsync's behaviour not to just ignore `-z` flags for local file transfers?

**Reply by WayneD** [reply]
Because you asked for it.  Why should it not do what you asked?

**Reply by jb2170** [reply]
It seems pointless and a waste of energy. What's to stop one from changing the behaviour from
`--compress, -z           compress file data during the transfer`
to
`--compress, -z           compress file data during the transfer if transferring over a network`

That would allow one to write `rsync -az "${SRC}" "${DEST}"` in a script and not care if say `SRC` is of the form `local-path/` or `user@host:~remote-path/`
